haiku-misc/gadgeteer/patches/gadgeteer-0.8.1.patchset

```diff
@@ -0,0 +1,558 @@
+-			WMSlider * owner) :
++			WMSlider *owner) :
+ 	BSlider(frame, "slider", title,
+ 			new BMessage('smup'), 0, 1000, B_BLOCK_THUMB,
+ 			B_FOLLOW_ALL_SIDES),
+@@ -165,7 +166,7 @@ WMSlider::WMSlider(BRect view_frame, int32 min, int32 max,
+ 	BRect sr(SLIDER_X_PAD, txtspace,
+ 		 view_bound.Width()-SLIDER_X_PAD - t->Bounds().Width(),
+ 		  txtspace+10);
+-	s = new _WMCntrlSldr(sr, NULL, t, min, max, valoc, this);
++	s = new _WMCntrlSldr(sr, NULL, t, (long int)min, (long int)max, (long int*)valoc, this);
+ 	s->ResizeTo(s->Bounds().Width(), s->ThumbFrame().Height()+2*SLIDER_Y_PAD);
+ 	b->AddChild(s);
+ 	t->MoveTo(s->Frame().RightTop());
+diff --git a/demo.sh b/demo.sh
+new file mode 100644
+index 0000000..52d6405
+--- /dev/null
++++ b/demo.sh
+@@ -0,0 +1,46 @@
++#! /bin/sh
++
++# A short script to illustrate the use of the "gadgeteer"
++# command-line app.
++
++# A function is needed to let the shell read the output:
++useOutput() {
++  while true; do
++	read TITLE
++	if [ $? != 0 ]; then
++	  return; fi
++	read FIRST
++	read LAST
++	read AGE
++	read CHECK
++	read BUTTON
++	if [ "$BUTTON" ]; then echo "You pressed the button"; continue; fi
++	echo "The name is $TITLE $FIRST $LAST"
++	echo "Age is $AGE"
++	if [ "$CHECK" ]; then echo "and Married"; fi
++  done
++}
++
++# Using shell variables can be convenient:
++  INIT="Billy"
++  
++# The demonstration command line:
++gadgeteer {title: DEMO, label: this is a demonstration, \
++     label: of some gadgeteer features, \
++     menu: [Mr, Mrs, Miss, Ms., Master, Sir, Lady, President, Their Majesty], \
++     width: 300, div: 80, pos: right, textcontrol: {label: First Name, init: $INIT}, \
++     pos: below, textcontrol: LastName, \
++     slider: {label: Age, min: 1, max: 110, init: 21}, \
++     checkbox: {label: Married, init: off}, \
++     button: {label: PUSH ME, width: 50}} | useOutput
++     
++# The command arguments are a YAML-syntax string and thus must be enclosed
++# in braces.  Each 'Directive' entry is a "key: value" pair, separated
++# by commas.  The space after a colon or comma is mandatory.
++
++# By default, items are stacked vertically.  The "pos: right" resets this
++# to put following items to the right of the first, the "pos: below" resets
++# again.  Note that the last item is vertically aligned with the previous.
++# The menu has a list of items as its value; most of the other controls have
++# extended parameter sets as their values.  The "div" global parameter
++# (currently) only applies to the textcontrol divider position.
+diff --git a/gadgeteer.rdef b/gadgeteer.rdef
+new file mode 100644
+index 0000000..5498710
+--- /dev/null
++++ b/gadgeteer.rdef
+@@ -0,0 +1,16 @@
++
++resource app_signature "application/x-vnd.pjg-gadgeteer";
++
++resource app_version {
++	major  = 0,
++	middle = 8,
++	minor  = 1,
++
++	variety = B_APPV_BETA,
++	internal = 0,
++
++	short_info = "Command Line GUI Interface",
++	long_info = "Display GUI controls from the command-line or shell-script\n"
```

Comment:
Maybe:
```diff
-+	long_info = "Display GUI controls from the command-line or shell-script\n"
++	long_info = "Display GUI controls from the command-line or shell-script"
```
(IDK if a full stop would be welcome.)

Hmm, if you think upstream might release a new version someday then it might be worth creating a `gadgeteer.rdef.in` in `additional-files/` and defining `ADDITIONAL_FILES`. But this is not a requirement.